overlord/configstate/configcore/hostname.go

```diff
@@ -41,7 +41,7 @@ func init() {
 // We are conservative here and follow hostname(7). The hostnamectl
 // binary is more liberal but let's err on the side of caution for
 // now.
-var validHostname = regexp.MustCompile(`^[a-z0-9][a-z0-9-]{1,62}$`).MatchString
+var validHostname = regexp.MustCompile(`^[a-z0-9][a-z0-9-]{1,62}(\.[a-z0-9-]{1,63})*$`).MatchString
```

Comment:
this doesn't seem to match a single letter hostname with no `.` such as in `a`. The man page doesn't seem to exclude this, should we support it?